src/Microsoft.Diagnostics.Runtime/Interfaces/IClrModule.cs

```diff
@@ -0,0 +1,132 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Diagnostics.Runtime.DacInterface;
+using System.Collections.Generic;
+using System.Diagnostics;
+
+namespace Microsoft.Diagnostics.Runtime.Interfaces
+{
+    /// <summary>
+    /// <see cref="ClrModule"/>
+    /// </summary>
+    public interface IClrModule
+    {
+        /// <summary>
+        /// <see cref="ClrModule.Address"/>
+        /// </summary>
+        ulong Address { get; }
+
+        /// <summary>
+        /// <see cref="ClrModule.AppDomain"/>
+        /// </summary>
+        ClrAppDomain AppDomain { get; }
+
+        /// <summary>
+        /// <see cref="ClrModule.AssemblyAddress"/>
+        /// </summary>
+        ulong AssemblyAddress { get; }
+
+
+        /// <summary>
+        /// <see cref="ClrModule.AssemblyName"/>
+        /// </summary>
+        string? AssemblyName { get; }
+
+        /// <summary>
+        /// <see cref="ClrModule.DebuggingMode"/>
+        /// </summary>
+        DebuggableAttribute.DebuggingModes DebuggingMode { get; }
+
+        /// <summary>
+        /// <see cref="ClrModule.ImageBase"/>
+        /// </summary>
+        ulong ImageBase { get; }
+
+        /// <summary>
+        /// <see cref="ClrModule.IsDynamic"/>
+        /// </summary>
+        bool IsDynamic { get; }
+
+        /// <summary>
+        /// <see cref="ClrModule.IsPEFile"/>
+        /// </summary>
+        bool IsPEFile { get; }
+
+        /// <summary>
+        /// <see cref="ClrModule.Layout"/>
+        /// </summary>
+        ModuleLayout Layout { get; }
```

Comment:
Need interface for ModuleLayout or is it a struct.

Reply:
It's an Enum